adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -666,6 +676,18 @@ def _instantiate_object_schema(
             self._relation_perceptions.append(
                 Relation(PART_OF, root_object_perception, sub_object_perception)
             )
+            # If the sub-object does not already have a color,
+            # use the parent's prototypical color if one exists
+            sub_object_color = re.findall(
```

Comment:
If there is a structured way to access data, use it rather than processing things as strings - it's much more robust.